synapse/config/registration.py

```diff
@@ -203,287 +203,12 @@ def generate_config_section(
         self, generate_secrets: bool = False, **kwargs: Any
     ) -> str:
         if generate_secrets:
-            registration_shared_secret = 'registration_shared_secret: "%s"' % (
+            registration_shared_secret = 'registration_shared_secret: %s' % (
                 random_string_with_symbols(50),
             )
+            return registration_shared_secret
```

Comment:
I don't think this is safe: the output of `random_string_with_symbols(50)` doesn't look like it's guaranteed to be a valid yaml string literal. 
- it might start with `&`, denoting a yaml anchor rather than a string
- it might start with `*`, denoting a yaml merge rather than a string
- other unlikely edge cases: all digits would be interpreted as a number. all colons seems to confuse https://yaml-online-parser.appspot.com/

```suggestion
            return f'registration_shared_secret: "{random_string_with_symbols(50)}"'
```